demo/src/screens/componentScreens/WithScrollEnablerScreen/AutoLockFlatList.tsx

```diff
@@ -0,0 +1,98 @@
+import _ from 'lodash';
+import React, {memo} from 'react';
+import {
+  FlatList,
+  FlatListProps,
+  StyleSheet,
+  LayoutChangeEvent
+} from 'react-native';
+import memoize from 'memoize-one';
+import {
+  Colors,
+  Text,
+  View,
+  withScrollEnabler,
+  WithScrollEnablerProps
+} from 'react-native-ui-lib';
+
+export type AutoLockScrollViewProps = FlatListProps<number> & {
+  numberOfItems: number;
+};
+
+const AutoLockFlatList = (props: AutoLockScrollViewProps) => {
+  const numberOfItems = props.numberOfItems;
+
+  const WithScrollEnabler = withScrollEnabler(
+    (props: WithScrollEnablerProps) => {
+      const getData = memoize((numberOfItems: number) => {
```

Comment:
even though you memoize this function you still re-creating it on each render.

Reply:
Changed to useCallback